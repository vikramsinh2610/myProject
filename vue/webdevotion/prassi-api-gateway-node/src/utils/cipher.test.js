const { encrypt, decrypt, ALGORITHMS } = require('./cipher');

describe('Cipher', () => {
  const text = 'pippo';
  const password = 'topolino';

  test('should encrypt a string', () => {
    const crypted = encrypt(text, password, ALGORITHMS.AES_256);
    expect(crypted).toMatchSnapshot();
  });

  test('should decrypt a string', () => {
    const crypted = encrypt(text, password, ALGORITHMS.AES_256);
    const decrypted = decrypt(crypted, password, ALGORITHMS.AES_256);
    expect(decrypted).toEqual('pippo');
  });
});
