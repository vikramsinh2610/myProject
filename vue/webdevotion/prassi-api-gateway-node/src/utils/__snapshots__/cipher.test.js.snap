// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cipher should encrypt a string 1`] = `"8cb0740467"`;
